### DISCUSSION TITLE
Migration changed for WorkspaceDb

### DISCUSSION THREAD

**Discussion by Angelk90** [zed-industries/zed#22023] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Every time I run `Zed` latest code from repository I get code like this printed.

What can I do so that I don't get these types of messages anymore.

```sql
[2024-12-13T20:35:04+01:00 ERROR Users] Migration changed for WorkspaceDb at step 14

Stored migration:
CREATE TABLE breakpoints (
  workspace_id INTEGER NOT NULL,
  worktree_path BLOB NOT NULL,
  relative_path BLOB NOT NULL,
  breakpoint_location INTEGER NOT NULL,
  FOREIGN KEY (workspace_id) REFERENCES workspaces (workspace_id) ON DELETE CASCADE ON UPDATE CASCADE
) STRICT;

Proposed migration:
ALTER TABLE
  panes
ADD
  COLUMN pinned_count INTEGER DEFAULT 0;
[2024-12-13T20:35:04+01:00 ERROR editor] buffer has no file
[2024-12-13T20:35:05+01:00 ERROR Users] Migration changed for WorkspaceDb at step 14

Stored migration:
CREATE TABLE breakpoints (
  workspace_id INTEGER NOT NULL,
  worktree_path BLOB NOT NULL,
  relative_path BLOB NOT NULL,
  breakpoint_location INTEGER NOT NULL,
  FOREIGN KEY (workspace_id) REFERENCES workspaces (workspace_id) ON DELETE CASCADE ON UPDATE CASCADE
) STRICT;

Proposed migration:
ALTER TABLE
  panes
ADD
  COLUMN pinned_count INTEGER DEFAULT 0;
[2024-12-13T20:35:05+01:00 ERROR Users] Migration changed for WorkspaceDb at step 14

Stored migration:
CREATE TABLE breakpoints (
  workspace_id INTEGER NOT NULL,
  worktree_path BLOB NOT NULL,
  relative_path BLOB NOT NULL,
  breakpoint_location INTEGER NOT NULL,
  FOREIGN KEY (workspace_id) REFERENCES workspaces (workspace_id) ON DELETE CASCADE ON UPDATE CASCADE
) STRICT;

Proposed migration:
ALTER TABLE
  panes
ADD
  COLUMN pinned_count INTEGER DEFAULT 0;

```

### Environment

```
Zed: v0.166.0 (Zed Preview)
OS: macOS 14.1.1
Memory: 16 GiB
Architecture: x86_64
```

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary>

```

```
</details>

**Comment by osiewicz** [comment]
`rm -rf ~/Library/Application\ Support/Zed/db/0-dev`
You've built Zed with debugger support and that interferes with the db layout that we have on latest main.

**Comment by Angelk90** [comment]
@osiewicz : Could you add the following suggestion here, I don't know based on the os used, that would be great.

https://github.com/zed-industries/zed/blob/9daa426e936c5d0b2f3a24e1cffde434c38a13de/crates/sqlez/src/migrations.rs#L72-L80

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
